src/document-converter.ts

```diff
@@ -1041,49 +1044,75 @@ export class DocumentConverter {
+          'shadycss/custom-style-interface.html',
+          'shadycss/entrypoints/custom-style-interface.js');
+    }
+    // Convert any ".html" URLs to point to their new ".js" module equivilent
+    jsUrl = replaceHtmlExtensionIfFound(jsUrl);
+    return jsUrl as ConvertedDocumentUrl;
+  }
+
+  /**
+   * Converts the URL for a script that is already being loaded in a
+   * pre-conversion HTML document via the <script> tag. This is similar to
+   * convertDocumentUrl(), but can skip some of the more complex .html -> .js
+   * conversion/rewriting.
+   */
+  private convertScriptUrl(oldUrl: OriginalDocumentUrl): ConvertedDocumentUrl {
+    // TODO(fks): This can be removed later if type-checking htmlUrl is enough
+    if (!isOriginalDocumentUrlFormat(oldUrl)) {
+      throw new Error(
+          `convertDocumentUrl() expects an OriginalDocumentUrl string` +
+          `from the analyzer, but got "${oldUrl}"`);
+    }
+    // Use the layout-specific UrlHandler to convert the URL.
+    return this.urlHandler.convertUrl(oldUrl);
+  }
+
+  /**
    * Format an import from the current document to the given JS URL. If an
    * original HTML import URL is given, attempt to match the format of that
    * import URL as much as possible. For example, if the original import URL was
    * an absolute path, return an absolute path as well.
+   *
+   * TODO(fks) 10-26-2017: Make this run on Windows/Non-Unix systems.
    */
   private formatImportUrl(
-      jsRootUrl: ConvertedDocumentUrl, originalHtmlImportUrl?: string) {
-    // Return an absolute URL path if the original HTML import was absolute
+      toUrl: ConvertedDocumentUrl, originalHtmlImportUrl?: string): string {
+    // Return an absolute URL path if the original HTML import was absolute.
+    // TODO(fks) 11-06-2017: Still return true absolute paths when using
+    // bare/named imports?
     if (originalHtmlImportUrl && path.posix.isAbsolute(originalHtmlImportUrl)) {
-      return '/' + jsRootUrl.slice('./'.length);
+      return '/' + toUrl.slice('./'.length);
     }
-    // TODO(fks): Most of these can be calculated once and saved for later
-    const isImportFromLocalFile =
-        !this.convertedUrl.startsWith('./node_modules');
-    const isImportToLocalFile = !jsRootUrl.startsWith('./node_modules');
-    const isPackageScoped = this.packageName.includes('/');
-    const isPackageElement = this.packageType === 'element';
-    let importUrl = getRelativeUrl(this.convertedUrl, jsRootUrl);
-    // If this document is an external dependency, or if this document is
-    // importing a local file, just return normal relative URL between the two
-    // files.
-    if (!isImportFromLocalFile || isImportToLocalFile) {
-      return importUrl;
+    // If the import is contained within a single package (internal), return
+    // a path-based import.
+    if (this.urlHandler.isImportInternal(this.convertedUrl, toUrl)) {
+      return this.urlHandler.getPathImportUrl(this.convertedUrl, toUrl);
```

Comment:
this is the same as like 1115, why the `if`?

Reply:
Yea this is weird, it comes from the addition of getNameImportUrl() in the next PR. Fixing...